memutils/memory.h

```diff
@@ -16,42 +16,72 @@
 #define MEMUTILS_MEMORY_H_
 
 #include <stddef.h>
+#include <stdint.h>
 #include "../cgoutils/utils.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+enum {
+  DEVICE_MEMORY_IMPLEMENTATION_FLAG = 1,
```

Comment:
HOST_MEMORY=0,
DEVICE_MEMORY=1,
DEVICE_POOLED_MEMORY=2?

Reply:
we will use each bit of the flag to tell some boolean values, so we will not use continue value like here.